library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -282,23 +330,44 @@ public void deactivate()
     store.disconnect();
   }
 
+  protected Object extractKey(ResultSet rs) throws SQLException
+  {
+    return rs.getObject(this.key);
+  }
+
   /**
    * Execute the query and transfer results to the emit queue.
    * @param preparedStatement PreparedStatement to execute the query and fetch results.
    */
-  protected void insertDbDataInQueue(PreparedStatement preparedStatement) throws SQLException, InterruptedException
+  protected int insertDbDataInQueue(PreparedStatement preparedStatement) throws SQLException, InterruptedException
   {
+    int resultCount = 0;
     preparedStatement.setFetchSize(getFetchSize());
     ResultSet result = preparedStatement.executeQuery();
     if (result.next()) {
       do {
-        while (execute && !emitQueue.offer(getTuple(result))) {
+        T obj = getTuple(result);
+        while (execute && !emitQueue.offer(obj)) {
```

Comment:
Not part of this PR but seems like emitQueue does not need to be a LinkedBlockingDeque since we are not using the deque feature of adding at both ends or removing from both.

Reply:
I modified the loop as there is no reason to check for result.next() first and then have a do .. while.